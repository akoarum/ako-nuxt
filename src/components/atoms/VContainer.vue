<template>
  <div :style="styled" class="vContainer">
    <slot />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  props: {
    maxWidth: { type: String }
  },
  computed: {
    styled(): { [index: string]: string } {
      const style = {}

      if (this.maxWidth) {
        style['--maxWidth'] = this.maxWidth
      }

      return style
    }
  }
})
</script>

<style lang="scss" scoped>
.vContainer {
  --maxWidth: 1024px;
  box-sizing: border-box;
  background: #fff;

  @include media-min {
    padding: 28px 16px 30px;
    max-width: var(--maxWidth);
  }

  @include media-max {
    padding: 28px 12px 30px;
  }
}
</style>
