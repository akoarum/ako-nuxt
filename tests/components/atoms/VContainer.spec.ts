import { shallowMount } from '@vue/test-utils'
import VContainer from '~/components/atoms/VContainer.vue'

describe('VContainer', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(VContainer, {
      propsData: {}
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styled: propsからスタイルを返す', () => {
      wrapper.setProps({ maxWidth: '1500px' })
      expect(wrapper.vm.styled).toEqual({ '--maxWidth': '1500px' })
    })
  })
})
