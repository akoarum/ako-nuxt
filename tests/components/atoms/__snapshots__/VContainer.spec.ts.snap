// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VContainer initialized snapshot 1`] = `
<div
  class="vContainer"
/>
`;
